src/utils/app/conversation.ts

```diff
@@ -392,40 +398,78 @@ export const saveConversations = (conversations: Conversation[]) => {
 //   }
 // }
 
-export async function saveConversationToServer(
+export async function saveConversationMetadata(
   conversation: Conversation,
   course_name: string,
 ) {
-  const MAX_RETRIES = 3
-  let retryCount = 0
+  const payload = convertChatToDBConversation(conversation)
```

Comment:
This basically strips off the messages from the conversation object here. And if you refer to what the handler receives, it still assumes it receives the messages in the payload.